stairway/src/main/java/bio/terra/stairway/impl/FlightFilterAccess.java

```diff
@@ -289,19 +277,70 @@ private void storeFlightPredicateValue(
    */
   @VisibleForTesting
   String makeInputPredicateSql(FlightFilterPredicate predicate) {
-    return "(I.key = '"
+    String valuePredicate = "";
+    if (predicate.getDatatype() == Datatype.NULL) {
+      return "(EXISTS (SELECT 0 FROM flightinput I WHERE F.flightid = I.flightid AND I.key = "
+          + "'"
+          + predicate.getKey()
+          + "' AND I.value IS NULL) "
+          + "OR NOT EXISTS (SELECT 0 FROM flightinput I WHERE F.flightid = I.flightid AND I.key = "
+          + "'"
+          + predicate.getKey()
+          + "'))";
+    } else if (predicate.getDatatype() == Datatype.LIST) {
+      valuePredicate =
+          "I.value "
+              + predicate.getOp().getSql()
+              + " (SELECT JSON_ARRAY_ELEMENTS_TEXT(CAST(:"
```

Comment:
This is pretty odd. What is this supposed to do?

Reply:
oh, that's in order to work with the JSON encoded data in the flight input table